www/js/survey/enketo/enketoHelper.ts

```diff
@@ -15,14 +19,133 @@ export type SurveyOptions = {
+  jsonDocResponse: string; //survey response as JSON object
+};
+
+type EnketoResponse = {
+  data: EnketoResponseData; //survey response data
+  metadata: any;
+};
+
+type EnketoSurveyConfig = {
+  [surveyName: string]: {
+    formPath: string;
+    labelTemplate: { [lang: string]: string };
+    labelVars: { [activity: string]: { [key: string]: string; type: string } };
+    version: number;
+    compatibleWith: number;
+  };
+};
+
+const LABEL_FUNCTIONS = {
+  UseLabelTemplate: async (xmlDoc: XMLDocument, name: string) => {
+    let configSurveys = await _lazyLoadConfig();
+
+    const config = configSurveys[name]; // config for this survey
+    const lang = i18next.resolvedLanguage;
+    const labelTemplate = config.labelTemplate?.[lang];
+
+    if (!labelTemplate) return 'Answered'; // no template given in config
+    if (!config.labelVars) return labelTemplate; // if no vars given, nothing to interpolate,
+    // so we return the unaltered template
+
+    // gather vars that will be interpolated into the template according to the survey config
+    const labelVars = {};
+    for (let lblVar in config.labelVars) {
+      const fieldName = config.labelVars[lblVar].key;
+      let fieldStr = _getAnswerByTagName(xmlDoc, fieldName);
+      if (fieldStr == '<null>') fieldStr = null;
+      if (config.labelVars[lblVar].type == 'length') {
+        const fieldMatches = fieldStr?.split(' ');
+        labelVars[lblVar] = fieldMatches?.length || 0;
+      } else {
+        throw new Error(`labelVar type ${config.labelVars[lblVar].type} is not supported!`);
+      }
+    }
+
+    // use MessageFormat interpolate the label template with the label vars
+    const mf = new MessageFormat(lang);
+    const label = mf.compile(labelTemplate)(labelVars);
+    return label.replace(/^[ ,]+|[ ,]+$/g, ''); // trim leading and trailing spaces and commas
+  },
+};
+
+/**
+ * _getAnswerByTagName look up how a question was answered, given the survey response
+ *   and the tag name of the question
+ * @param {XMLDocument} xmlDoc survey response as XML object
+ * @param {string} tagName tag name of the question
+ * @returns {string} answer string. If not found, return "\<null\>"
+ */
+function _getAnswerByTagName(xmlDoc: XMLDocument, tagName: string) {
+  const vals = xmlDoc.getElementsByTagName(tagName);
+  const val = vals.length ? vals[0].innerHTML : null;
+  if (!val) return '<null>';
+  return val;
+}
+
+/** @type {EnketoSurveyConfig} _config */
+let _config: EnketoSurveyConfig;
+
+/**
+ * _lazyLoadConfig load enketo survey config. If already loaded, return the cached config
+ * @returns {Promise<EnketoSurveyConfig>} enketo survey config
+ */
+export function _lazyLoadConfig() {
+  if (_config !== undefined) {
+    return Promise.resolve(_config);
+  }
+  return getConfig().then((newConfig) => {
+    logInfo('Resolved UI_CONFIG_READY promise in enketoHelper, filling in templates');
+    _config = newConfig.survey_info.surveys;
+    return _config;
+  });
+}
```

Comment:
After analyzing a bit further, I think this function is pointless.
`getConfig()` is already cached/lazyloaded, so we could just call it directly. The only difference is that `getConfig()` returns the whole config obj while this function only returns the `survey_info.surveys` sub-field of the config obj.

I think it would be clearer to remove this function. Anywhere it was used, call `getConfig()` and access the subfields from there.
Then move your `EnketoSurveyConfig` type def to `js/types/appConfigTypes.ts` (it doesn't exist yet on this branch but I started it on your other PR https://github.com/e-mission/e-mission-phone/pull/1098)
The reasoning is that eventually we're going to want a single source of truth for the entire config – not scattered typings for each sub-field of the config.

Reply:
I made those changes in [d123b4f](https://github.com/e-mission/e-mission-phone/pull/1063/commits/d123b4f23f90537671e13e59ff672f7156e655c2), including moving the type. I did not work the type into the existing appConfig type yet though